### PR TITLE
Also remove https repos in zypper_clear_repos

### PR DESCRIPTION
**Pull request by DrMullings**
This time with a regex in the original find expression

- Related ticket: https://progress.opensuse.org/issues/66259

Followup to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10578
Verification needed

### CODE REVIEW

File: tests/update/zypper_clear_repos.pm [hunk 1]
```diff
@@ -26,7 +26,7 @@ sub run {
     my $repos_folder = '/etc/zypp/repos.d';
     zypper_call 'lr -d', exitcode => [0, 6];
     assert_script_run(
-"find $repos_folder/*.repo -type f -exec grep -q 'baseurl=http://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
+"find $repos_folder/*.repo -type f -exec grep -Eq 'baseurl='[http|https]://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
```
Comment (Vogtinator) [inline tests/update/zypper_clear_repos.pm:29]:
```suggestion
"find $repos_folder/*.repo -type f -exec grep -Eq 'baseurl='(http|https)://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
```

**Comment by DrMullings** [comment]
Thanks for merging, but maybe we should've verified it first?

**Comment by Vogtinator** [comment]
> Thanks for merging, but maybe we should've verified it first?

It's not merged yet - without a verification run and success from travis that wouldn't be a good idea. The approval rating is only for code review AFAIK.

**Comment by DrMullings** [comment]
Confused the merged badge from the old one, sorry that

File: tests/update/zypper_clear_repos.pm (hunk 1 above)
Comment (DrMullings) [reply tests/update/zypper_clear_repos.pm]: Adapted

File: tests/update/zypper_clear_repos.pm [hunk 2]
```diff
@@ -26,7 +26,7 @@ sub run {
     my $repos_folder = '/etc/zypp/repos.d';
     zypper_call 'lr -d', exitcode => [0, 6];
     assert_script_run(
-"find $repos_folder/*.repo -type f -exec grep -q 'baseurl=http://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
+"find $repos_folder/*.repo -type f -exec grep -Eq 'baseurl='(http|https)://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
```
Comment (Vogtinator) [inline tests/update/zypper_clear_repos.pm:29]:
```suggestion
"find $repos_folder/*.repo -type f -exec grep -Eq 'baseurl=(http|https)://download.opensuse.org/' {} \\; -delete && echo 'unneed_repos_removed' > /dev/$serialdev",
```

Comment (foursixnine) [reply tests/update/zypper_clear_repos.pm:29]: +1 was just gonna comment the same

**Comment by DrMullings** [comment]
https://openqa.opensuse.org/tests/1327407

https://openqa.opensuse.org/tests/1327408

Removed the Quote and started one NET and one DVD verification

**Comment by DrMullings** [comment]
> Still works locally, though now without hypothetical false positives.
> 
> Maybe matching `'baseurl=[^:]+://download.opensuse.org/'` is a better option, though currently we mostly use HTTP(s) 

Wouldn't this match everything before ':' ? I would prefer keeping it as an OR, but I am no regex expert

My openqa-clone-custom-git-refspec seems still broken though, so if someone could run

```
openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10661 https://openqa.opensuse.org/tests/1327002 TEST=jrauch_test GROUP=0 _SKIP_POST_FAIL_HOOKS=1
openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10661 https://openqa.opensuse.org/tests/1325341 TEST=jrauch_test GROUP=0 _SKIP_POST_FAIL_HOOKS=1
```
we can verify it on O3

**Comment by Vogtinator** [comment]
Done:

https://openqa.opensuse.org/t1327410
https://openqa.opensuse.org/t1327411

**Comment by Vogtinator** [comment]
> > Still works locally, though now without hypothetical false positives.
> > Maybe matching `'baseurl=[^:]+://download.opensuse.org/'` is a better option, though currently we mostly use HTTP(s)
> 
> Wouldn't this match everything before ':' ? I would prefer keeping it as an OR, but I am no regex expert

Yes, so also `rsync://`, `ftps://`, `htTpS://` or just `baseurl="http://`.

**Comment by DrMullings** [comment]
Both verification are fine, travis has no complaints, can we merge this or do you insist on matching different protocols other than http(s)?

**Comment by Vogtinator** [comment]
> Both verification are fine, travis has no complaints, can we merge this or do you insist on matching different protocols other than http(s)?

I'm fine with either.

**Comment by DrMullings** [comment]
Then merge it please

**Pull request merged**